### ISSUE TITLE
Released XGBoost models don't match the released taxonomy's encoding (expect 954/1104 features; taxonomy encodes to 462)

### ISSUE THREAD

**Issue by Paul-Kyle** [jenna-russell/storyscope#2]
First — thank you for releasing code, taxonomy, features, *and* weights. Very few papers ship all four, and the release was complete enough that we could both reproduce your headline result and diagnose this mismatch precisely.

## Summary

The pretrained models in `data/models/` cannot be used with the released `data/taxonomy.json`: the boosters expect 954 (narrative) / 1104 (full) input features, but the shipped taxonomy encodes to 462 columns under `encode_features(mode="multi_hot")` — the mode `train_classifier.py` uses. No encoding scheme we tried reconciles them.

## Reproduction

```python
import xgboost as xgb
from storyscope.utils.feature_encoder import (
    load_taxonomy, build_feature_type_map, get_taxonomy_feature_ids, encode_features)
import pandas as pd

tax  = load_taxonomy("data/taxonomy.json")
fmap = build_feature_type_map(tax)
fids = get_taxonomy_feature_ids(tax)
ref  = pd.read_parquet("data/storyscope_features.parquet").rename(columns={"source": "author"})

X, cols = encode_features(ref, fids, fmap, mode="multi_hot")
print(X.shape)  # (61575, 462)

clf = xgb.XGBClassifier(); clf.load_model("data/models/binary_narrative.json")
print(clf.get_booster().num_features())  # 954
```

## Encoding schemes tested against the shipped taxonomy (304 features: 124 categorical, 59 ordinal, 45 scale, 44 binary, 32 multi-select w/ 190 total values)

| Scheme | Columns |
|---|---|
| label-encode singles + multi-hot multi-selects (`multi_hot`) | 462 |
| + count column per multi-select (`multi_hot_count`) | 494 |
| one-hot categorical, label rest | 864 |
| one-hot categorical+ordinal, label rest | 1011 |
| one-hot everything | 1234 |
| **models expect** | **954 / 1104** |

Our best guess: the weights were trained against a pre-stage-4c (pre-deduplication) taxonomy that wasn't included in the release.

## Workaround (works fine)

Retraining from the shipped feature matrix with your own `train_classifier.py` is self-consistent and reproduces the paper: binary macro-F1 **0.9617** on `storyscope_features.parquet` (paper reports 0.932; we didn't test multiclass). So nothing is lost scientifically — but anyone who loads the shipped weights directly will hit a silent feature-count mismatch or garbage predictions.

**Suggested fix:** either include the taxonomy the models were trained against, or refresh `data/models/` from the released taxonomy + features.

## PS — minor robustness note in stage 5

`apply_features.py`'s per-dimension call assumes `generate_json` returns a dict; with `response_mime_type="application/json"` the model occasionally returns a top-level **list**, which the `result.items()` path converts into a silently dropped dimension. We hit this ~1/200 calls with `gemini-3-flash-preview`. An `isinstance(parsed, dict)` check that routes wrong shapes into the retry loop fixes it.

---

Context: we hit this while applying StoryScope's feature space to output from our own narrative-generation system (multi-agent game transcripts). Happy to provide more detail on any of the above.
